src/Assets.php

```diff
@@ -57,6 +60,29 @@ public static function register_assets() {
 			$asset_api->register_script( 'wc-blocks-checkout', 'build/blocks-checkout.js', [] );
 		}
 
+		// Pass in settings specific to blocks.
+		$product_counts = wp_count_posts( 'product' );
+
+		$asset_data_registry->add( 'wcBlocksPluginUrl', plugins_url( '/', __DIR__ ) );
+		$asset_data_registry->add( 'wcBlocksPhase', Package::feature()->get_flag() );
+
+		/*
+		 * translators: If your word count is based on single characters (e.g. East Asian characters),
+		 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
+		 * Do not translate into your own language.
+		 */
+		$asset_data_registry->add( 'wordCountType', _x( 'words', 'Word count type. Do not translate!', 'woo-gutenberg-products-block' ) );
+		$asset_data_registry->add( 'placeholderImgSrc', wc_placeholder_img_src() );
+		$asset_data_registry->add( 'isLargeCatalog', $product_counts->publish > 100 );
+		$asset_data_registry->add( 'productCount', array_sum( (array) $product_counts ) );
+		$asset_data_registry->add(
+			'restApiRoutes',
+			[
+				'/wc/store' => array_keys( Package::container()->get( RestApi::class )->get_routes_from_namespace( 'wc/store' ) ),
+			]
+		);
```

Comment:
The idea was we'd include more niche settings here (for blocks to consume) without polluting _core settings_ and causing future bw compat issues. Comment aside, are you still ok with these being located here?

Reply:
Yes I understand that. But from a consumer perspective, given this plugin is included in core as a package, these values are always going to be present anytime the settings global is output for a route because the data is being added on `init`. So effectively, what you're aiming for here isn't happening. Any route that outputs `wc-settings` is going to have this data (unless I'm missing something).  

If we truly want to distinguish these values from "core" settings, then we'd want to only register this data when any block is rendered (as opposed to when registering block scripts). 

Also, since the keys of the settings aren't namespaced that well (`productCount` for instance), we could nest those settings inside an associative array representing blocks and then just expose the entire object as a registered setting? Something like:

```js
$woo_blocks_data = [
    'wordCountType' => _x( 'words', 'Word count type. Do not translate!', 'woo-gutenberg-products-block' ) ),
    'placeholderImgSrc' => wc_placeholder_image_src(),
    // etc...
];
$asset_data_registry->add( 'woo-blocks-data', $woo_blocks_data );
```

Reply:
I'm not sure what direction to take here then. These could all be defined in the core settings section but that seems wrong for block specific settings. Could we somehow populate settings with a closure and avoid window globals? I'm not sure if that's possible or worth. Ideally we should be able to add private namespaced settings for blocks without worrying about direct access and BW compatibility, because the window global bypasses the API completely.